climate_emotions_map/make_descriptive_plots.py

```diff
@@ -209,85 +266,164 @@ def make_impact_plot_traces(df: pd.DataFrame):
+        x = data_category[COL_DEMOGRAPHIC_VARIABLE].apply(
+            get_demographic_variable_to_display
+        )
         traces.append(
             go.Bar(
-                x=data_category[COL_DEMOGRAPHIC_VARIABLE].apply(
-                    get_demographic_variable_to_display
+                x=x.apply(lambda x: wrap_text_label(x, width=text_wrap_width)),
+                y=data_category[COL_PERCENTAGE] * 100,
+                # text=[
+                #     f"{percentage*100:.1f}% ({n})"
+                #     for percentage, n in zip(
+                #         data_category[COL_PERCENTAGE], data_category[COL_N]
+                #     )
+                # ],
+                text=x.apply(
+                    lambda x: wrap_text_label(x, width=text_wrap_width)
                 ),
-                y=data_category[COL_N],
-                customdata=data_category[COL_PERCENTAGE] * 100,
+                customdata=list(zip(x, data_category[COL_N])),
                 hovertemplate=(
-                    "<b>%{x}</b>"
-                    f"<br>{category}: %{{y}} (%{{customdata:.1f}}%)"
+                    "<b>%{customdata[0]}</b>"
+                    ": %{y:.2f}% (%{customdata[1]})"
                     "<extra></extra>"
                 ),
+                marker_color=marker_color,
+                hoverinfo="none",
                 name=category,
             )
         )
     return traces
 
 
-def make_descriptive_plots(state: str | None = None) -> go.Figure:
+def make_descriptive_plots(
+    state: str | None = None, margins=None, text_wrap_width=14
+) -> go.Figure:
+
+    if margins is None:
+        margins = {"l": 0, "r": 0, "t": 20, "b": 0}
 
     # get data to plot
     if state is None:
         data = SAMPLEDESC_WHOLESAMPLE
     else:
         data = SAMPLEDESC_STATE.loc[SAMPLEDESC_STATE["state"] == state]
 
+    row_heights = [
+        n_categories + 0.5
+        for n_categories in [2, 2, 3, 2, 3, 2, 3, 3, 5, 5, 8]
+    ]
+
     # initialize figure
     fig = make_subplots(
-        rows=6,
-        cols=6,
-        specs=[
-            [{"colspan": 2}, None, {"colspan": 2}, None, {"colspan": 2}, None],
-            [{"colspan": 3}, None, None, {"colspan": 3}, None, None],
-            [{"colspan": 3}, None, None, {"colspan": 3}, None, None],
-            [{"colspan": 3}, None, None, {"colspan": 3}, None, None],
-            [{"colspan": 6}, None, None, None, None, None],
-            [{"colspan": 6}, None, None, None, None, None],
-        ],
+        rows=len(row_heights),
+        cols=1,
+        row_heights=row_heights,
         subplot_titles=[
             get_demographic_variable_to_display(demographic_variable)
-            for demographic_variable in SUBPLOT_POSITIONS.keys()
+            for demographic_variable in SUBPLOT_POSITIONS
         ],
+        vertical_spacing=0.04,
+    )
+
+    colorscale_step = 1 / (len(row_heights) - 1)
+    colors = sample_colorscale(
```

Comment:
I like your default colorscale a lot - maybe expose as a list in case Eric has specific preferences?